### DISCUSSION TITLE
Building QField on Windows 10: Failed to install qca-qt6

### DISCUSSION THREAD

**Discussion by jzk-star** [opengisch/QField#5353] [Q&A]
### Describe the issue
I encountered some issues while setting up the environment to build QField on Windows 10.

### Reproduction steps
1. I am using CMake version 3.29.5, vcpkg version vcpkg2023-07-19, and QField version 3.0.5.
2. According to the development documentation, the CMake command I used to build is:
cmake -S QField -B build -D VCPKG_TARGET_TRIPLET=x64-windows-static  -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=D:/build" -D PKG_CONFIG_EXECUTABLE="C:/src/vcpkg/packages/pkgconf_x64-windows/tools/pkgconf/pkgconf.exe"
3. An error occurred during the build process at qca-qt6:x64-windows-static. The screenshot is as follows:
![image](https://github.com/opengisch/QField/assets/83906896/5228e59d-fc8a-41ee-ac06-a057ec2ba07f)
The log file screenshots are as follows:
![image](https://github.com/opengisch/QField/assets/83906896/7f5e4ab2-9b9a-4c67-afc4-4db15147df02)
![image](https://github.com/opengisch/QField/assets/83906896/db58f9cb-6748-48f2-a6d4-23184c911735)

### Expected behavior
Could you please suggest possible solutions for this issue?
One additional point: I chose vcpkg version vcpkg2023-07-19 because higher versions of vcpkg use Python 3.11 for installing QField, which always results in errors. Therefore, I opted for a lower version of vcpkg.
Could you update the development documentation to include the specific CMake and vcpkg versions required for the setup? This would help avoid unnecessary issues during installation.

**Comment by m-kuhn** [comment]
According to the docs, the command is slightly different from what you describe
https://github.com/opengisch/QField/blob/master/doc/dev.md#configure-1

```
cmake -S QField -B build \
  -D VCPKG_TARGET_TRIPLET=x64-windows-static \
  -D CMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded\$<\$<CONFIG:Debug>:Debug> \
  -D PKG_CONFIG_EXECUTABLE=build/vcpkg_installed/x64-windows/tools/pkgconf/pkgconf.exe \
  -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:/build"
```

One thing I note here is that probably `x64-windows` should be `x64-windows-static` though.

Could you try with the documented instructions instead?

**Comment by jzk-star** [comment]
This time, I strictly followed the commands provided in the development documentation, but it still failed![image](https://github.com/opengisch/QField/assets/83906896/17bb8678-df11-4c59-b688-221aa595fe52)
I tried to make some modifications to the commands in the documentation, mainly changing x64-windows to x64-windows-static. I copied the build/vcpkg_installed/x64-windows/tools/pkgconf folder to build/vcpkg_installed/x64-windows-static/tools/pkgconf and modified -D PKG_CONFIG_EXECUTABLE=build/vcpkg_installed/x64-windows/tools/pkgconf/pkgconf.exe to -D PKG_CONFIG_EXECUTABLE=build/vcpkg_installed/x64-windows-static/tools/pkgconf/pkgconf.exe.

Additionally, I removed the two \ in -D CMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded\$<\$<CONFIG:Debug>:Debug>.

Finally, the command I modified is: cmake -S QField -B build -D VCPKG_TARGET_TRIPLET=x64-windows-static -D CMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>" -D PKG_CONFIG_EXECUTABLE="build/vcpkg_installed/x64-windows-static/tools/pkgconf/pkgconf.exe" -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:/build".

I successfully installed a series of dependencies, but at the end of the build, some unexpected errors still occurred, as shown in the screenshots below:
![image](https://github.com/opengisch/QField/assets/83906896/89a37bab-4c82-4cb9-a018-e0b5eb0e4072)
![image](https://github.com/opengisch/QField/assets/83906896/068addd7-2c17-47ef-91a1-0bbd594d6e2d)
I found similar issues in previous issues, but no solutions have been provided yet. Screenshots are as follows:
![image](https://github.com/opengisch/QField/assets/83906896/4fb706ba-d363-438d-8488-76379536bbd1)

**Reply by m-kuhn** [reply]
It looks like it's almost done and failing in the very last step.
I am not completely sure what's wrong there, this will need a bit of investigation on cmake side.

I would suggest to also ask in the existing discussion and to leave a link to this discussion in here.

**Reply by jzk-star** [reply]
I am temporarily unable to find the cause of the error, so I tried to build another release version of QField.
I changed the QField version from 3.0.5 to 3.3.2.
However, higher versions of QField often correspond to higher versions of vcpkg. When using vcpkg versions greater than vcpkg2023-07-19, the Python version used during the build process changes to Python 3.11 (previously it was Python 3.10 when building QField 3.0.5). During the installation of dependencies with vcpkg, I always encounter the error "python ImportError: DLL load failed while importing _socket" (this information is read from the log).
The command I used is still cmake -S QField -B build -D VCPKG_TARGET_TRIPLET=x64-windows-static -D CMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$CONFIG:Debug:Debug>" -D PKG_CONFIG_EXECUTABLE="build/vcpkg_installed/x64-windows-static/tools/pkgconf/pkgconf.exe" -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=D:/build".
The error screenshot is as follows:
![image](https://github.com/opengisch/QField/assets/83906896/6ff51848-cb79-43f3-ad6d-3a63155eb498)
The content of the error log file is as follows:
![image](https://github.com/opengisch/QField/assets/83906896/09432176-c054-439b-9b63-46b5b68381d5)
![image](https://github.com/opengisch/QField/assets/83906896/14098aab-a920-4c27-9922-2dcd366ab00b)

**Comment by m-kuhn** [comment]
I don't know what your ultimate goal is with QField, but for the record: it's now also possible to customize it through plugins. https://www.opengis.ch/2024/06/18/supercharge-your-fieldwork-with-qfields-project-and-app-wide-plugins/

**Reply by jzk-star** [reply]
Using QField is intended for mobile development.

**Reply by nirvn** [reply]
@jzk-star , you should try to see if your needs can be fulfilled by writing a plugin for QField instead of going through the burden of self-compilation (as well as the follow up maintenance burden of keep up to date with releases).

If you are exploring the plugin option, let us know how it goes!
